File: app/src/main/java/com/polije/proyektopupin/Register/RegisterActivity.java
package com.polije.proyektopupin.Register;

import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.view.WindowManager;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;

import com.google.android.material.textfield.TextInputEditText;
import com.polije.proyektopupin.Login.LoginActivity;
import com.polije.proyektopupin.R;

public class RegisterActivity extends AppCompatActivity {


    //Variables
    TextInputEditText txt_user, txt_email, txt_pass, txt_confirm;
    Button btn_regis, btngoogle;
    TextView signIn;
    String username, email, password, confirmpass;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);
        setContentView(R.layout.activity_register);

        if(getSupportActionBar() !=null){
            getSupportActionBar().hide();
        }

        //Hooks
        txt_user = findViewById(R.id.txt_username);
        txt_email = findViewById(R.id.txt_email);
        txt_pass = findViewById(R.id.txt_password);
        txt_confirm = findViewById(R.id.txt_confirmpass);
        btn_regis = findViewById(R.id.btn_register);
        btngoogle = findViewById(R.id.btn_google);
        signIn = findViewById(R.id.signup);

        //Aktivity ke login
        signIn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                AlertDialog.Builder builder = new AlertDialog.Builder(RegisterActivity.this);
                builder.setMessage("Already Have an Account ?");

                //ketika tombol ya ditekan
                builder.setPositiveButton("Yes", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        Intent intent = new Intent( RegisterActivity.this, LoginActivity.class);
                        startActivity(intent);
                        finish();

                        //menampilkan toast
                        Toast.makeText(RegisterActivity.this, "Come On, Make You Activity Now !", Toast.LENGTH_SHORT).show();
                    }
                });

                //ketika tombol tidak ditekan
                builder.setNegativeButton("No", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        dialog.dismiss();
                    }
                });

                //Menampilkan dialog
                AlertDialog alertDialog = builder.create();
                alertDialog.show();
            }
        });

        btn_regis.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                String usernameVal = txt_user.getText().toString();
                String emailVal = txt_email.getText().toString();
                String passwordVal = txt_pass.getText().toString();
                String confirmVal = txt_confirm.getText().toString();

                if (usernameVal.isEmpty() || emailVal.isEmpty() || passwordVal.isEmpty() || confirmVal.isEmpty()){
                    //show text if column empty
                    Toast.makeText(RegisterActivity.this, "Please Fill the Column First", Toast.LENGTH_SHORT).show();
                    
                } else {
                    
                    if(!emailVal.contains("@")){
                        //show text if email doesn't sign @
                        Toast.makeText(RegisterActivity.this, "Your Email Doesn't Have the @ Sign", Toast.LENGTH_SHORT).show();

                    } else if(!passwordVal.trim().equals(confirmVal.trim())) {
                        //show text if password and confirm password not same
                        Toast.makeText(RegisterActivity.this, "Password and Confirm Password Must Be the Same", Toast.LENGTH_SHORT).show();

                    } else {
                        //show text the account has been registered
                        Toast.makeText(RegisterActivity.this, "Your New Account Has Been Registered", Toast.LENGTH_SHORT).show();
                    }
                }
            }
        });

    }
}

File: app/src/main/java/polije/kuliah/topupin/presentation/user/RegisterActivity.kt
package polije.kuliah.topupin.presentation.user

import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import jakarta.inject.Inject
import polije.kuliah.topupin.R
import polije.kuliah.topupin.data.model.User
import polije.kuliah.topupin.data.model.UserRegister
import polije.kuliah.topupin.databinding.ActivityRegisterBinding
import polije.kuliah.topupin.presentation.Injector

class RegisterActivity: AppCompatActivity() {

    @Inject
    lateinit var factory: UserViewModelFactory

    private lateinit var userViewModel: UserViewModel
    private lateinit var binding: ActivityRegisterBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        (application as Injector).createHomeSubComponent().inject(this)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_register)

        userViewModel = ViewModelProvider(this, factory)
            .get(UserViewModel::class.java)

        binding.btnRegister.setOnClickListener{
            val usernameText = binding.txtUsername.text.toString()
            val emailText = binding.txtEmail.text.toString()
            val passText = binding.txtPassword.text.toString()
            val ConfirmpassText = binding.txtConfirmpass.text.toString()
            val noHpText = binding.txtTelepon.text.toString()
            val namas = binding.txtNama.text.toString()

            if (usernameText.isEmpty() || emailText.isEmpty() || passText.isEmpty() || ConfirmpassText.isEmpty()) {
                Toast.makeText(this, "Semua kolom wajib diisi", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            if (passText != ConfirmpassText) {
                Toast.makeText(this, "Password dan Konfirmasi Password tidak cocok", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            userViewModel.registerUser(UserRegister(usernameText, passText, emailText, noHpText, namas))
            Toast.makeText(this, "Registrasi berhasil!", Toast.LENGTH_SHORT).show()
        }
    }
}